packages/wrangler/src/deploy/open-next.ts

```diff
@@ -0,0 +1,52 @@
+import { runCommand } from "../autoconfig/c3-vendor/command";
+import { getInstalledPackageVersion } from "../autoconfig/frameworks";
+import { logger } from "../logger";
+import { getPackageManager } from "../package-manager";
+
+/**
+ * Checks whether a project is an OpenNext one and in such case delegates the deployment to `@opennextjs/cloudflare`.
+ * If the project is not an OpenNext one, or the `OPEN_NEXT_DEPLOY` environment variable is set (meaning that `wrangler
+ * deploy` has been called from OpenNext itself), then no delegation happens.
+ *
+ * Why?
+ *  `wrangler deploy` shouldn't be run on an open-next project, since open-next has its own deploy command,
+ *  in the interest in having things "just work" we updated `wrangler deploy` to delegate to the open-next
+ *  deploy command, so that, if a developer runs `npx wrangler deploy` (e.g. like in Workers Builds) on an
+ *  open-next project that will work just fine.
+ *
+ * @param projectRoot The path to the project's root
+ * @returns true is the deployment has been delegated to open-next, false otherwise
+ */
+export async function maybeDelegateToOpenNextDeployCommand(
+	projectRoot: string | undefined
+): Promise<boolean> {
+	const openNextVersion =
+		projectRoot &&
+		getInstalledPackageVersion("@opennextjs/cloudflare", projectRoot);
+
+	const isOpenNextProject = openNextVersion !== undefined;
+
+	if (isOpenNextProject) {
+		// eslint-disable-next-line turbo/no-undeclared-env-vars
+		const openNextDeploy = process.env["OPEN_NEXT_DEPLOY"];
```

Comment:
Is there a reason this doesn't use the existing Wrangler helpers for env vars in `misc-variables.ts`?